tensorflow/compiler/xla/service/gpu/gpu_executable.cc

```diff
@@ -476,18 +504,24 @@ StatusOr<ExecutionOutput> GpuExecutable::ExecuteAsyncOnStream(
     absl::optional<HloInputOutputAliasConfig::Alias> alias =
         input_output_alias.GetAliasedParameter(index);
     if (alias) {
-      CHECK_LT(alias->parameter_number, arguments.size());
-      ExecutionInput& input = arguments[alias->parameter_number];
-      MaybeOwningDeviceMemory* maybe_owning_memory =
-          input.MutableBuffer(alias->parameter_index);
-      if (alias->must_alias() && !maybe_owning_memory->HasOwnership()) {
+      MaybeOwningDeviceMemory* maybe_owning_memory = nullptr;
+      // ScopedBuffer is never an owned buffer.
+      if (auto* args0 = absl::get_if<absl::Span<const ShapedBuffer* const>>(&arguments)) {
+        CHECK_LT(alias->parameter_number, args0->size());
+      } else {
+        auto args1 = absl::get<1>(arguments);
```

Comment:
Same naming/temp var comments as before.

Reply:
For the naming yes. But for the temp, I can't do it as the variable is used twice. One for the CHECK_LT and one to get the good inputs.